tests/spark/autologging/conftest.py

```diff
@@ -2,34 +2,33 @@
 import tempfile
 
 import pytest
-from pyspark.sql import Row, SparkSession
+from pyspark.sql import Row
 from pyspark.sql.types import IntegerType, StringType, StructField, StructType
 
-from tests.spark.autologging.utils import _get_mlflow_spark_jar_path
+from mlflow._spark_autologging import clear_table_infos
+
+from tests.spark.autologging.utils import _get_or_create_spark_session
 
 
 @pytest.fixture(scope="module")
 def spark_session():
-    jar_path = _get_mlflow_spark_jar_path()
-    with SparkSession.builder.config("spark.jars", jar_path).master(
-        "local[*]"
-    ).getOrCreate() as session:
+    with _get_or_create_spark_session() as session:
         yield session
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture
 def data_format(format_to_file_path):
     res, _ = sorted(format_to_file_path.items())[0]
     return res
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture
```

Comment:
Changing scope to function so as not to share same file paths across test, which might hide some failure.
For example, SparkAutologgingContext accumulates events to global variable `_table_info`, so those events can be carried over to the other test cases, unless cleaned explicitly (like I added in tear_down fixture). However, the unit tests have been passing so far because the file path in the event was same.